lightning/src/offers/offer.rs

```diff
@@ -163,7 +163,10 @@ impl OfferBuilder {
 	///
 	/// Successive calls to this method will override the previous setting.
 	pub fn supported_quantity(mut self, quantity: Quantity) -> Self {
-		self.offer.supported_quantity = quantity;
+		self.offer.supported_quantity = match quantity {
+			Quantity::Max(n) if n.get() == 1 => Quantity::One,
```

Comment:
Wondering if I should instead have only two variants (`Quantity::Max(n)` and `Quantity::Many`) and default to `Quantity::Max(1)`. Then would translate to the appropriate `Option<u64>` internally for the TLV stream. Thoughts?

Reply:
FYI, I changed this to two variants `Bounded(n)` and `Unbounded` as discussed offline.